sdk/src/main/java/com/google/cloud/dataflow/sdk/io/BoundedSource.java

```diff
@@ -133,6 +133,95 @@ public Double getFractionConsumed() {
     }
 
     /**
+     * A constant to use as the return value for {@link #getParallelismConsumed()} or
+     * {@link #getParallelismRemaining()} when the exact value is unknown.
+     */
+    public static final long PARALLELISM_UNKNOWN = -1;
```

Comment:
Consumed parallelism: clearly integral.

Remaining paralellism: use is exhorted to only populate if exact. Hence also clearly integral.

For the user API, these concepts are integral and neither of these can be `Inf`.

Of course the interface between the job and the Dataflow service need not follow the user API.